File: Sliding Window/219. Contains Duplicate II/219. Contains Duplicate II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        sett = set()
        for i, n in enumerate(nums):
            if n in sett:
                return True
            sett.add(n)
            if len(sett) > k:
                sett.remove(nums[i - k])
        return False

# Time: O(n)
# Space: O(k)

File: Sliding Window/219. Contains Duplicate II/219. Contains Duplicate II.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function (nums, k) {
  let sett = new Set();
  for (let i = 0; i < nums.length; i++) {
    if (sett.has(nums[i])) return true;
    sett.add(nums[i]);
    if (sett.size > k) sett.delete(nums[i - k]);
  }
  return false;
};

// Time: O(n)
// Space: O(k)
